ledger/sandbox-on-x/src/test/lib/scala/com/daml/platform/sandbox/auth/GrantPermissionTest.scala

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.sandbox.auth
+
+import scala.concurrent.Future
+import com.daml.ledger.api.v1.admin.{user_management_service => ums}
+import com.daml.test.evidence.scalatest.ScalaTestSupport.Implicits._
+
+import java.util.UUID
+
+trait GrantPermissionTest {
+
+  this: AdminOrIDPAdminServiceCallAuthTests with UserManagementAuth =>
+
+  val adminPermission =
+    ums.Right(ums.Right.Kind.ParticipantAdmin(ums.Right.ParticipantAdmin()))
+
+  val idpAdminPermission =
+    ums.Right(ums.Right.Kind.IdentityProviderAdmin(ums.Right.IdentityProviderAdmin()))
+
+  def serviceCallWithGrantPermission(
```

Comment:
The name of this method gave me a pause, because I thought that it was being invoked using a token with a given permission, suggest a small naming change to make it clear:
```suggestion
  def serviceCallToGrantPermission(
```